server/src/main/java/org/elasticsearch/common/settings/ListKeyStoreCommand.java

```
@@ -46,13 +47,19 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
         if (keystore == null) {
             throw new UserException(ExitCodes.DATA_ERROR, "Elasticsearch keystore not found. Use 'create' command to create one.");
         }
-
-        keystore.decrypt(new char[0] /* TODO: prompt for password when they are supported */);
-
-        List<String> sortedEntries = new ArrayList<>(keystore.getSettingNames());
-        Collections.sort(sortedEntries);
-        for (String entry : sortedEntries) {
-            terminal.println(entry);
+        char[] passphrase;
+        passphrase = keystore.hasPassword() ? KeyStoreWrapper.readPassphrase(terminal, false) : new char[0];
+        try {
+            keystore.decrypt(passphrase);
+            List<String> sortedEntries = new ArrayList<>(keystore.getSettingNames());
+            Collections.sort(sortedEntries);
+            for (String entry : sortedEntries) {
+                terminal.println(entry);
+            }
+        } catch (SecurityException e) {
+            throw new UserException(ExitCodes.DATA_ERROR, "Failed to access the keystore. Please make sure the passphrase was correct.");
```

lets not swallow this exception; we should add them to the `UserException`